packages/mongo/src/lib/MongoProvider.ts

```diff
@@ -635,7 +648,7 @@ export class MongoProvider<StoredValue = unknown> extends JoshProvider<StoredVal
 	}
 
 	public async [Method.Size](payload: SizePayload): Promise<SizePayload> {
-		payload.data = (await this.collection.countDocuments({})) || 0;
+		payload.data = (await this.collection.countDocuments({})) || payload.data;
```

Comment:
If this returns `undefined` or `null` then use nullish operator.
```suggestion
		payload.data = (await this.collection.countDocuments({})) ?? payload.data;
```